### PR TITLE
Pass through exit code from monitored process to callers of LogMonitor

### LINKED ISSUE #36: LogMonitor does not pass through the exit code from a monitored process so calling scripts cannot act on errors
As the title states.

LogMonitor logs the errors from the monitored application but does not pass them on to any callers. This is different from errors that occur while it tries to set up the monitoring, those are returned to the caller.

### PR DESCRIPTION
**Pull request by ulfaxelssoncab**
Fixes #36. Did not see the white space thing in VS...

@algamaes @manoj-kadam

### CODE REVIEW

**Comment by profnandaa** [comment]
@ulfaxelssoncab --  thanks for your contribution. We're onboarding on to the project as maintenance; once we are done with the house-keeping, this will be reviewed promptly.

File: LogMonitor/src/LogMonitor/Main.cpp [hunk 1]
```diff
@@ -230,6 +230,7 @@ int __cdecl wmain(int argc, WCHAR *argv[])
 {
     std::wstring cmdline;
     PWCHAR configFileName = (PWCHAR)DEFAULT_CONFIG_FILENAME;
+    int exitcode = 0;
```
Comment (bobsira) [inline LogMonitor/src/LogMonitor/Main.cpp:203]: Probably not a big issue but we can have this as ExitCode in future @iankingori

File: LogMonitor/src/LogMonitor/Main.cpp [hunk 2]
```diff
@@ -288,7 +289,7 @@ int __cdecl wmain(int argc, WCHAR *argv[])
             cmdline += argv[i];
         }
 
-        CreateAndMonitorProcess(cmdline);
+        exitcode = CreateAndMonitorProcess(cmdline);
```
Comment (bobsira) [inline LogMonitor/src/LogMonitor/Main.cpp:271]: It looks like the type of CreateAndMonitorProcess function is DWORD and exitcode is int. 

I reckon we need to do some type conversion and error checks if possible?

File: LogMonitor/src/LogMonitor/ProcessMonitor.cpp [hunk 3]
```diff
@@ -65,22 +65,20 @@ DWORD CreateAndMonitorProcess(std::wstring& Cmdline)
...
 /// \param Cmdline      The command to start the new process.
 ///
-void CreateChildProcess(std::wstring& Cmdline)
+DWORD CreateChildProcess(std::wstring& Cmdline)
 {
     PROCESS_INFORMATION piProcInfo;
     STARTUPINFO siStartInfo;
     bool bSuccess = false;
-    DWORD status;
+    DWORD exitcode = 0;
```
Comment (bobsira) [inline LogMonitor/src/LogMonitor/ProcessMonitor.cpp:88]: Declaration of the variable should suffice.

**Review by bobsira — COMMENTED** [review]
The change looks good.

File: LogMonitor/src/LogMonitor/Main.cpp (hunk 2 above)
Comment (iankingori) [reply LogMonitor/src/LogMonitor/Main.cpp:271]: DWORD to int conversion is handled implicitly, open to running a few tests to see chances of failure

**Review by iankingori — APPROVED** [review]

File: LogMonitor/src/LogMonitor/Main.cpp (hunk 1 above)
Comment (iankingori) [reply LogMonitor/src/LogMonitor/Main.cpp:203]: Any reason for the capital letters or should this be `exit_code`?

**Review by bobsira — APPROVED** [review]

File: LogMonitor/src/LogMonitor/Main.cpp (hunk 1 above)
Comment (bobsira) [reply LogMonitor/src/LogMonitor/Main.cpp:203]: I'll have to confirm with the C++ conventions for this, but I guess it comes down to preference at this point

**Pull request merged**